lib/rules/amd-function-arity.js

```diff
@@ -30,10 +31,41 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
+        if (allowExtraDependencies) {
+            return false;
+        }
+
+        if (context.options && context.options[0] && context.options[0].allowExtraDependencies) {
+            return context.options[0].allowExtraDependencies;
+        }
+
+        return false;
+    }
+
     function requireHasCallback(node) {
         return node.arguments.filter(isFunctionExpression).length;
     }
 
+    function getExtraPaths(dependencyNodes, callbackParams) {
+        var extraPaths = [];
+        var extraPathQty = dependencyNodes.length - callbackParams.length;
+
+        for (var i = dependencyNodes.length - 1; i >= dependencyNodes.length - extraPathQty; i--) {
+            extraPaths.push(dependencyNodes[i].value);
+        }
+        return extraPaths;
+    }
+
+    function allPathsAllowed(paths, allowedPaths) {
+        for (var i = 0; i < paths.length; i++) {
```

Comment:
Similar to above, I think this could be simplified with `Array.prototype.every`.